src/libraries/Microsoft.PowerFx.Core/Public/Types/FormulaType.cs

```diff
@@ -233,5 +234,19 @@ public override int GetHashCode()
         #endregion // Equality
 
         public abstract void Visit(ITypeVisitor vistor);
+
+        internal virtual void DefaultExpressionValue(StringBuilder sb)
+        {
+            throw new NotSupportedException();
```

Comment:
>NotSupportedException [](http://example.com/codeflow?start=22&length=21)

nit:
we can improve the error message here by including the derived type. 

$"{this.GetType().FullName} doesn't implement DefaultExpressionValue"